relay/polkadot/src/lib.rs

```diff
@@ -1780,12 +1832,94 @@ pub type Migrations = (migrations::Unreleased, migrations::Permanent);
 /// The runtime migrations per release.
 #[allow(deprecated, missing_docs)]
 pub mod migrations {
+	use polkadot_runtime_common::traits::Leaser;
+
 	use super::*;
 
 	parameter_types! {
 		pub const ImOnlinePalletName: &'static str = "ImOnline";
 	}
 
+	pub struct GetLegacyLeaseImpl;
+	impl coretime::migration::GetLegacyLease<BlockNumber> for GetLegacyLeaseImpl {
+		fn get_parachain_lease_in_blocks(para: ParaId) -> Option<BlockNumber> {
+			let now = frame_system::Pallet::<Runtime>::block_number();
+			let lease = slots::Leases::<Runtime>::get(para);
+			if lease.is_empty() {
+				return None
+			}
+			// Lease not yet started/or having holes, refund (coretime can't handle this):
+			if lease.iter().any(Option::is_none) {
+				if let Err(err) = slots::Pallet::<Runtime>::clear_all_leases(
+					frame_system::RawOrigin::Root.into(),
+					para,
+				) {
+					log::error!(
+						target: "runtime",
+						"Clearing lease for para: {:?} failed, with error: {:?}",
+						para,
+						err
+					);
+				};
+				return None
+			}
+			let (index, _) =
+				<slots::Pallet<Runtime> as Leaser<BlockNumber>>::lease_period_index(now)?;
+			Some(index.saturating_add(lease.len() as u32).saturating_mul(LeasePeriod::get()))
+		}
+	}
+
+	/// Enable the elastic scaling node side feature.
+	///
+	/// This is required for Coretime to ensure the relay chain processes parachains that are
+	/// assigned to multiple cores.
+	pub struct EnableElasticScalingNodeFeature;
+	impl OnRuntimeUpgrade for EnableElasticScalingNodeFeature {
+		fn on_runtime_upgrade() -> Weight {
+			use runtime_parachains::configuration::WeightInfo as _;
+			let _ = Configuration::set_node_feature(frame_system::RawOrigin::Root.into(), 1, true);
+			weights::runtime_parachains_configuration::WeightInfo::<Runtime>::set_node_feature()
+		}
+	}
```

Comment:
no longer needed, thanks to https://collectives.subsquare.io/fellowship/referenda/173